react/layout.jsx

```diff
@@ -45,13 +54,15 @@ let Layout = React.createClass({
         yield dattcore.asyncInitialize()
 
         let userSetupFlag = yield dattcore.asyncGetUserSetupFlag()
-        if (!userSetupFlag && (window.location.hash === '#/frontpage' || !window.location.hash || window.location.hash === '#/')) {
+
+        if (!userSetupFlag && !this.state.skipSetup && (window.location.hash === '#/frontpage' || !window.location.hash || window.location.hash === '#/') && (window.location.toString().indexOf('tests.html') === -1)) {
```

Comment:
This is probably fine, but I'm uneasy about using "window" for routing, because it doesn't work in node.js. The reason we would want the UI to work in node.js is so we can render it server-side before sending it. If there is a better way to handle routing that doesn't involve "window" we should use that.